src/Plugin/Commerce/PaymentGateway/Heartland.php

```diff
@@ -0,0 +1,391 @@
+use GlobalPayments\Api\Services\ReportingService;
+
+/**
+ * Provides the On-site payment gateway.
+ *
+ * @CommercePaymentGateway(
+ *   id = "commerce_heartland",
+ *   label = "Heartland",
+ *   display_label = "Token-based payment gateway from Heartland",
+ *   forms = {
+ *     "add-payment-method" = "Drupal\commerce_heartland\PluginForm\Onsite\PaymentMethodAddForm",
+ *   },
+ *   payment_method_types = {"credit_card"},
+ *   credit_card_types = {
+ *     "amex", "dinersclub", "discover", "jcb", "maestro", "mastercard", "visa",
+ *   },
+ *   js_library = "commerce_heartland/form"
+ * )
+ */
+
+
+class Heartland extends OnsitePaymentGatewayBase implements OnsiteInterface
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, PaymentTypeManager $payment_type_manager, PaymentMethodTypeManager $payment_method_type_manager, TimeInterface $time)
+    {
+        parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $payment_type_manager, $payment_method_type_manager, $time);
+  
+        // Heartland SDK - Authentication
+        // https://developer.heartlandpaymentsystems.com/Documentation/authentication/#authentication
+        $config = new ServicesConfig();
+        $config->secretApiKey = $this->configuration['secret_key'];
+        $config->serviceUrl = "https://cert.api2.heartlandportico.com";
+        ServicesContainer::configure($config);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPublicKey()
+    {
+        // Used by PaymentMethodAddForm to send key to javascript
+        return $this->configuration['public_key'];
+    }
+  
+    /**
+     * {@inheritdoc}
+     */
+    public function defaultConfiguration()
+    {
+        // Store Card Data will be implemented in iteration 2 multi-use tokenization.
+        return [
+            'public_key' => '',
+            'secret_key' => '',
+            // 'store_card_data' => 'FALSE',
+        ] + parent::defaultConfiguration();
+    }
+  
+    /**
+     * {@inheritdoc}
+     */
+    public function buildConfigurationForm(array $form, FormStateInterface $form_state)
+    {
+        $form = parent::buildConfigurationForm($form, $form_state);
+  
+        // Matching schema in config/schema/commerce_heartland.schema.yml.
+        $form['public_key'] = [
+            '#type' => 'textfield',
+            '#title' => $this->t('Public Key'),
+            '#default_value' => $this->configuration['public_key'],
+            '#required' => true,
+        ];
+
+        $form['secret_key'] = [
+            '#type' => 'textfield',
+            '#title' => $this->t('Secret Key'),
+            '#description' => $this->t('Get your keys at <a href="https://developer.heartlandpaymentsystems.com/Account/KeysandCredentials" target="_blank">https://developer.heartlandpaymentsystems.com/Account/KeysandCredentials</a>.'),
```

Comment:
Let's change the verbiage on this link to something like `the Heartland developer portal` instead of the address itself.